cypress/e2e/awx/resources/jobTemplateSurveys.cy.ts

```diff
@@ -12,119 +12,414 @@ describe('Job Templates Surveys', function () {
-      //Assert job output screen
-      //Wait for job completion
-      //Assert survey info on job details screen
+      cy.intercept('PATCH', awxAPI`/job_templates/${jobTemplate.id.toString()}/`).as(
+        'enableSurvey'
+      );
+      cy.get('[for="survey-switch"]').click();
+      cy.wait('@enableSurvey');
+
+      cy.intercept('GET', awxAPI`/job_templates/${jobTemplate.id.toString()}/launch/`).as(
+        'launchTemplate'
+      );
+
+      cy.clickButton('Launch template');
+      cy.wait('@launchTemplate');
+
+      cy.contains('Prompt on Launch');
+
+      surveyTypes.forEach((survey) => {
+        const groupType = `survey-${survey.type}-answer-form-group`;
+
+        cy.getByDataCy(groupType).within(() => {
+          cy.contains(survey.question_name);
+          cy.contains('*');
+          cy.get('.pf-v5-c-icon').click();
+        });
+        cy.contains(survey.question_description);
+
+        if (['multiplechoice', 'multiselect'].includes(survey.type)) {
+          if (survey.type === 'multiplechoice') {
+            cy.getByDataCy(groupType).within(() => {
+              cy.contains(survey.default);
+              cy.get('div[data-ouia-component-id="menu-select"]').click();
+              survey?.choices?.forEach((choice) => {
+                cy.getByDataCy('survey-multiplechoice-answer').contains(choice);
+              });
+            });
+          } else {
+            cy.getByDataCy(groupType).within(() => {
+              const defaults = survey.default.split('\n');
+              defaults.forEach((defaultValue) => {
+                cy.contains(defaultValue);
+              });
+              cy.get('#survey-multiselect-answer').click();
+            });
+
+            cy.get('#survey-multiselect-answer-select').within(() => {
+              survey?.choices?.forEach((choice) => {
+                cy.getByDataCy(choice);
+              });
+            });
+          }
+        } else if (survey.type === 'password') {
+          cy.getByDataCy(groupType)
+            .getByDataCy('survey-password-answer')
+            .should('have.value', '$encrypted$');
+        } else {
+          cy.getByDataCy(groupType)
+            .getByDataCy(`survey-${survey.type.toLowerCase()}-answer`)
+            .should('have.value', survey.default);
+        }
+      });
+
+      cy.clickButton('Next');
+
+      cy.getByDataCy('code-block-value').within(() => {
+        surveyTypes.forEach((survey) => {
+          cy.contains(survey.variable);
+          if (survey.type === 'password') cy.contains('$encrypted$');
+          else {
+            survey.default.split('\n').forEach((def) => {
+              cy.contains(def);
+            });
+          }
+        });
+      });
+
+      cy.clickButton('Finish');
+
+      cy.intercept('POST', awxAPI`/job_templates/${jobTemplate.id.toString()}/launch/`).as(
+        'postLaunch'
+      );
+      cy.clickButton(/^Finish/);
+      cy.wait('@postLaunch')
+        .its('response.body.id')
```

Comment:
At this point in the test, I am seeing this error:
![image](https://github.com/ansible/ansible-ui/assets/63685497/bdf4385c-f8c9-4f4b-b7d8-d787c2d30108)

Reply:
yes, this was caused by https://github.com/ansible/ansible-ui/pull/2241, rerunning and should be fixed now